riscv/runtime/src/coprocessors.rs

```diff
@@ -5,8 +5,16 @@
 // during the reachability analysis.
 extern "C" {
     fn poseidon_coprocessor(a: u32, b: u32) -> u32;
+    fn poseidon_gl_coprocessor(a: PoseidonGLU256, b: PoseidonGLU256, c: PoseidonGLU256) -> PoseidonGLU256;
```

Comment:
these are the same arguments as the Poseidon co-processor: 8 field elements as input and 4 cap field elements

Reply:
What if instead, the list of values was passed as two parameters: `*mut u64` and a `usize` ? The first parameter points to the buffer or a `[u64]` memory slice. That wouldn't fix the register stuff just yet, but it would ensure that it only uses 2 arguments (which I believe is more stable, register wise, than using more arguments) and it's future proof in the sense that you could pass more than 2 field elements in the future (i.e. with a `usize` parameter > 24).

Why 24 ? because in the current API, I assume the first 8 bytes are for field element `a`, the next 8 are for field element `b` and the last 8 are for the output (hence the `mut` pointer).

Reply:
Re last assumption, it's actually 12 u64 input elements, and 4 output.

But yea, the buffer idea sounds good. Wouldn't it be even better if we use a `mut [u64; 12]` (if possible) on the Rust side? Because the PIL co-processor takes exactly 12 inputs, so I think we should ensure the same API. I'll try that next

Reply:
right, 12 and not 24.